lib/diff.js

```diff
@@ -117,18 +117,14 @@ const getAction = ({actual, ideal}) => {
   const binsExist = ideal.binPaths.every((path) => existsSync(path))
 
   // top nodes, links, and git deps won't have integrity, but do have resolved
-  // However! if *neither* has a resolved value, then we need to refetch
-  // to figure out what it even is, so we mark as CHANGE.
+  // if neither node has integrity, the bins exist, and either neither
+  // node has a resolved value or they both do and match, then we can
+  // leave this one alone since we already know the versions match due to
+  // the condition above
   const noIntegrity = !ideal.integrity && !actual.integrity
-  const resolvedMatch = ideal.resolved && ideal.resolved === actual.resolved
-  if (noIntegrity && resolvedMatch && binsExist)
-    return null
-
-  // If NEITHER has resolved OR integrity, but the versions match
-  // (which we know because of the check above), and we don't have
-  // to find bins, then we should just leave it as-is.
   const noResolved = !ideal.resolved && !actual.resolved
-  if (noIntegrity && noResolved && binsExist)
+  const resolvedMatch = ideal.resolved && ideal.resolved === actual.resolved
+  if (noIntegrity && binsExist && (resolvedMatch || noResolved))
```

Comment:
Wait a minute... I think maybe the comment was saying the right thing, and the code was wrong.

If we don't have any resolved, don't we still need to refetch?  Otherwise we'll never get a resolved value for it?

Reply:
AHH, no, because shrinkwraps might be incomplete.  Verified with the code.  I'll update the comment to prevent us thinking it's a mistake again ;)